api/client/swarm/opts.go

```diff
@@ -99,27 +101,31 @@ func (o *AutoAcceptOption) String() string {
 }
 
 // Set sets a new value on this option
-func (o *AutoAcceptOption) Set(value string) error {
-	value = strings.ToUpper(value)
-	switch value {
-	case "", "NONE":
-		if accept, ok := o.values[WORKER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", WORKER)
-		}
-		if accept, ok := o.values[MANAGER]; ok && accept {
-			return fmt.Errorf("value NONE is incompatible with %s", MANAGER)
-		}
-		o.values[WORKER] = false
-		o.values[MANAGER] = false
-	case WORKER, MANAGER:
-		if accept, ok := o.values[value]; ok && !accept {
-			return fmt.Errorf("value NONE is incompatible with %s", value)
+func (o *AutoAcceptOption) Set(acceptValues string) error {
+	for _, value := range strings.Split(acceptValues, ",") {
+		value = strings.ToUpper(value)
+		switch value {
+		case "", NONE:
+			// NONE must stand alone, so if any non-NONE setting exists, conflict
+			if !o.values[NONE] && len(o.values) > 0 {
+				var s []string
+				for r := range o.values {
+					s = append(s, r)
+				}
+				return fmt.Errorf("value NONE is incompatible with %s", strings.Join(s, ", "))
+			}
+			o.values[WORKER] = false
+			o.values[MANAGER] = false
```

Comment:
This seems wrong because elsewhere we're looking at `len(o.values)`. Setting a value to false won't change the number of values in the map. We should be deleting the value instead.

The real problem is that the type of the map is `map[string]bool`. It should be `map[string]struct{}` instead, because this map is being used as a set. Making that change will prevent this kind of error.